molten-http-client/src/main/java/com/hotels/molten/http/client/OkHttpCallFactoryFactory.java

```diff
@@ -141,4 +146,32 @@ private void addEventTracing(OkHttpClient.Builder clientBuilder, RetrofitService
             clientBuilder.eventListenerFactory(DelegatingEventListener.delegatedFactory(eventListenerFactories)::apply);
         }
     }
+
+    private List<Protocol> getProtocol(List<Protocols> protocol) {
+        List<Protocol> result;
+        result = protocol
+            .stream()
+            .map(this::convertToHttpProtocol)
+            .distinct()
+            .collect(Collectors.toList());
+        return result;
+    }
+
+
+    private Protocol convertToHttpProtocol(Protocols protocol) {
+        Protocol result;
+
+        if (protocol == Protocols.HTTP_2) {
```

Comment:
Have you considered to move this conversion into freshly introduced enum? There you could easily guarantee unequivocal mapping for both OkHttp and Netty protocols by introducing two variable.

Reply:
Done, please check if you are ok with this new approach  :)